src/pages/index.js

```diff
@@ -54,69 +60,229 @@ const cardSection = new Section(
+
+confirmDelete.setEventListeners();
+
+//Project 9 /////////////////////////////////////////////////////////////////////
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c801e776-9008-430b-a7ca-bcf7d2aaaf7f", // my special token
+    "Content-Type": "application/json",
+  },
+});
+
+// api
+//   .getInitialCards()
+//   .then((data) => {
+//     cardSection.renderItems(data);
+//     console.log(data);
+//     // console.log("Fetched Cards:", data);
+//   })
+//   .catch((err) => {
+//     console.error(err);
+//   });
+
+api
+  .getUserInfoAndCards()
+  .then(({ userInfo, cards }) => {
+    userInfor.setUserInfo({
+      name: userInfo.name,
+      description: userInfo.about,
+    });
+    userInfor.updateAvatarImage({ avatar: userInfo.avatar });
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Failed to load user information or cards:", err);
+  });
+
+// // Open avatar edit popup
+const avatarEditButton = document.querySelector(".profile__edit-avatar-button");
+avatarEditButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// // Function to handle avatar update
+function handleAvatarSubmit(inputData, saveButton) {
+  renderSaving(true, saveButton);
+  api
+    .updateUserAvatar({ avatar: inputData.avatar })
+    .then((userData) => {
+      userInfor.updateAvatarImage({ avatar: userData.avatar });
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      renderSaving(false, saveButton);
+    });
+}
+
+function renderSaving(isLoading, buttonElement, defaultText = "Save") {
+  if (isLoading) {
+    buttonElement.textContent = "Saving...";
+  } else {
+    buttonElement.textContent = defaultText;
+  }
+}
+
+//////////////////////////////////// end of project 9 ////////////////////////////////////////////
+
+// //----------------------------------------------------------------------------------------
 //                                     Functions
 //----------------------------------------------------------------------------------------
 function createCard(item) {
-  const card = new Card(item, cardSelector, handleImagePreview);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImagePreview,
+    (cardId, card) => {
+      deleteCardModal(cardId, card);
+    },
+    (cardId, card) => {
+      handleCardLike(cardId, card);
+    }
+  );
   return card.getview();
 }
 
+// function createCard(item) {
+//   const card = new Card(
+//     item,
+//     cardSelector,
+//     handleImagePreview,
+//     handleCardLike,
+//     (cardId, card) => {
+//       deleteCardModal(cardId, card);
+//     },
+//     (cardId, isLiked, cardElement) => {
+//       handleLikeIcon(cardId, isLiked, cardElement, card);
+//     }
+//   );
+//   return card.getview();
+// }
+
+function handleCardLike(card) {
+  if (!card._isLiked) {
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.